src/libs/kata-sys-util/src/k8s.rs

```diff
@@ -49,23 +51,68 @@ pub fn is_host_empty_dir(path: &str) -> bool {
     false
 }
 
-// update_ephemeral_storage_type sets the mount type to 'ephemeral'
+// patchup_container_storage_type to correct the mount type
+pub fn patchup_container_storage_type(oci_spec: &mut Spec) -> Result<()> {
```

Comment:
This patch splits the handling of volume into two processes: 1. obtaining the direct volume mount type, and 2. handling the volume. As a result, mountInfo needs to be parsed twice. I prefer to directly judge the direct volume in the `volume/mod.rs`. This way, mountInfo only needs to be parsed once and it is clear that direct volume is being processed.

Reply:
Thx @quanweiZhou Nice suggestion!
PR is updated now, Please help check it again.